fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -240,6 +249,26 @@ private void insertField(DocumentEntry entry, final Document document) {
         }
     }
 
+    public <M extends Message> String extractGrouping(@Nullable FDBIndexableRecord<M> record) {
+        KeyExpression expression = state.index.getRootExpression();
+        if (expression instanceof GroupingKeyExpression) {
+            final List<DocumentEntry> groupingExpressions = getFields(((GroupingKeyExpression)expression).getGroupingSubKey(), record, record.getRecord());
+            if (groupingExpressions.isEmpty()) {
+                return null;
+            }
+            String indexPreface = "";
+            List<Key.Evaluated> evaluatedList = ((GroupingKeyExpression)expression).getGroupingSubKey().evaluateMessage(record, record.getRecord());
+            for (Key.Evaluated key : evaluatedList) {
+                Object value = key.values().get(0);
```

Comment:
This is taking the first column of every row, while I believe it wants to take every column of the first row.

The contract is to only get a single grouping key for each record. Disallowing, for instance, a map-like repeated nested message where every value is _separately_ indexed according to its key. Which is probably okay for right now. So using `evaluateSingleton` might be appropriate.